src/data/polynomial/ring_division.lean

```diff
@@ -163,6 +163,20 @@ begin
   exact degree_le_mul_left p h2.2,
 end
 
+lemma eq_zero_of_dvd_of_degree_lt {p q : R[X]} (h₁ : p ∣ q) (h₂ : degree q < degree p) :
+  q = 0 :=
+begin
+  by_contradiction hc,
+  exact (lt_iff_not_ge _ _ ).mp h₂ (degree_le_of_dvd h₁ hc),
+end
+
+lemma eq_zero_of_nat_degree_lt {p q : R[X]} (h₁ : p ∣ q) (h₂ : nat_degree q < nat_degree p) :
```

Comment:
If I am not confused by the implications, you can use this to generalize [polynomial.monic.not_dvd_of_degree_lt](https://leanprover-community.github.io/mathlib_docs/data/polynomial/monic.html#polynomial.monic.not_dvd_of_degree_lt) removing the monic assumption. In doing so you should change the name of [polynomial.monic.not_dvd_of_degree_lt](https://leanprover-community.github.io/mathlib_docs/data/polynomial/monic.html#polynomial.monic.not_dvd_of_degree_lt), if this means modifying a lot of files where it is used can you please open another PR and asking for my review? It can be merged very quickly.

Reply:
Sure, I'll try and have a go at that later on today!

Reply:
Haha [polynomial.monic.not_dvd_of_degree_lt](https://leanprover-community.github.io/mathlib_docs/data/polynomial/monic.html#polynomial.monic.not_dvd_of_degree_lt) was added by me days ago, so not many things depend on it. It holds for any semiring without need of no_zero_divisor though. Since this section has `no_zero_divisors` present, you can indeed prove a version without `monic` here.

Reply:
Good point! Well I've just added the other version without `monic` - it's probably good to have both anyway

Reply:
But I guess in that case no need to remove `polynomial.monic.not_dvd_of_degree_lt` from the files that use it :)

Reply:
Yes, I didn't realize they had orthogonal assumptions.